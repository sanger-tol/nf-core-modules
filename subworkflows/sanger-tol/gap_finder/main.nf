//
// GENERATE BED FILE OF GAPS AND LENGTH IN REFERENCE
//

include { SEQTK_CUTN    } from '../../../modules/nf-core/seqtk/cutn/main'
include { GAWK          } from '../../../modules/nf-core/gawk/main'

workflow GAP_FINDER {
    take:
    ch_reference     // Channel [ val(meta), path(fasta) ]

    main:
    ch_versions     = Channel.empty()


    //
    // MODULE: GENERATES A GAP SUMMARY FILE
    //
    SEQTK_CUTN (
        ch_reference
    )
    ch_versions     = ch_versions.mix( SEQTK_CUTN.out.versions )


    //
    // MODULE: ADD THE LENGTH OF GAP TO BED FILE - INPUT FOR PRETEXT MODULE
    //
    GAWK (
        SEQTK_CUTN.out.bed,
        [],
        false
    )
    ch_versions     = ch_versions.mix( GAWK.out.versions )


    emit:
    gap_file        = GAWK.out.output
    versions        = ch_versions
}
