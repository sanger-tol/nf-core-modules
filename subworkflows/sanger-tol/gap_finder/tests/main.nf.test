nextflow_workflow {

    name "Test Subworkflow GAP_FINDER"
    script "../main.nf"
    config "./nextflow.config"
    workflow "GAP_FINDER"

    tag "subworkflows"
    tag "subworkflows_sangertol"
    tag "subworkflows/gap_finder"
    tag "seqtk/cutn"
    tag "gawk"
    tag "subworkflows/../../modules/nf-core/gawk"
    tag "subworkflows/../../modules/nf-core/seqtk/cutn"
	tag "modules/nf-core/gunzip"

    setup {
        nfcoreInitialise("${launchDir}/library/")
        nfcoreInstall(
            "${launchDir}/library/",
            [
                "seqtk/cutn",
                "gawk",
                "gunzip"
            ]
        )
        nfcoreLink("${launchDir}/library/", "${baseDir}/modules/")

        run("GUNZIP"){
            script "../../../../modules/nf-core/gunzip/main.nf"
            process {
                """
                input[0] = [
                    [ id: "Cloeon_Dipeterum" ],
                    file(params.modules_testdata_base_path + 'Cloeon_dipterum/assembly/release/ieCloDipt1.1/insdc/GCA_949628265.1.fasta.gz', checkIfExists: true)
                ]
                """
            }
        }
    }

    test("Cloeon_dipterum genome [fasta]") {

        when {
        
            params {
                gawk_args = "'BEGIN { OFS = \"\\t\" }{print \$0, sqrt((\$3-\$2)*(\$3-\$2))}'"
                seqtk_args = "-n 1"
            }


            workflow {
                """
                input[0] = GUNZIP.out.gunzip
                """
            }
        }

        then {
            assertAll(
                { assert workflow.success },
                { assert snapshot(workflow.out).match() }
            )
        }

        cleanup {
            nfcoreUnlink("${launchDir}/library/", "${baseDir}/modules/nf-core")
        }
    }

    test("Cloeon_dipterum genome - stub") {
        options "-stub"
        when {

            workflow {
                """
                input[0] = GUNZIP.out.gunzip
                """
            }
        }

        then {
            assertAll(
                { assert workflow.success },
                { assert snapshot(workflow.out).match() }
            )
        }

        cleanup {
            nfcoreUnlink("${launchDir}/library/", "${baseDir}/modules/nf-core")
        }
    }
}
