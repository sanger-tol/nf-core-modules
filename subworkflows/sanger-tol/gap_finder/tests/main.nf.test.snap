{
    "Cloeon_dipterum genome - stub": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "Cloeon_Dipeterum"
                        },
                        "Cloeon_Dipeterum.gap.bedgraph:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "1": [
                    "versions.yml:md5,929f4176dedc9856781d33262611dc9b",
                    "versions.yml:md5,cdedf30584c2f81dd4f1268f2371159d"
                ],
                "gap_file": [
                    [
                        {
                            "id": "Cloeon_Dipeterum"
                        },
                        "Cloeon_Dipeterum.gap.bedgraph:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "versions": [
                    "versions.yml:md5,929f4176dedc9856781d33262611dc9b",
                    "versions.yml:md5,cdedf30584c2f81dd4f1268f2371159d"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "25.04.7"
        },
        "timestamp": "2025-10-06T16:13:51.003190181"
    },
    "Cloeon_dipterum genome [fasta]": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "Cloeon_Dipeterum"
                        },
                        "Cloeon_Dipeterum.gap.bedgraph:md5,25a0c7406035024fac941b1e900a0bd4"
                    ]
                ],
                "1": [
                    "versions.yml:md5,929f4176dedc9856781d33262611dc9b",
                    "versions.yml:md5,cdedf30584c2f81dd4f1268f2371159d"
                ],
                "gap_file": [
                    [
                        {
                            "id": "Cloeon_Dipeterum"
                        },
                        "Cloeon_Dipeterum.gap.bedgraph:md5,25a0c7406035024fac941b1e900a0bd4"
                    ]
                ],
                "versions": [
                    "versions.yml:md5,929f4176dedc9856781d33262611dc9b",
                    "versions.yml:md5,cdedf30584c2f81dd4f1268f2371159d"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "24.04.4"
        },
        "timestamp": "2025-10-08T12:07:48.83352935"
    }
}